dotnet/src/SemanticKernel/AI/OpenAI/Services/AzureTextEmbeddings.cs

```diff
@@ -47,8 +48,15 @@ public async Task<IList<Embedding<float>>> GenerateEmbeddingsAsync(IList<string>
     {
         var deploymentName = await this.GetDeploymentNameAsync(this._modelId);
         var url = $"{this.Endpoint}/openai/deployments/{deploymentName}/embeddings?api-version={this.AzureOpenAIApiVersion}";
-        var requestBody = Json.Serialize(new AzureEmbeddingRequest { Input = data });
 
-        return await this.ExecuteEmbeddingRequestAsync(url, requestBody);
+        IEnumerable<Embedding<float>> embeddings = new List<Embedding<float>>();
```

Comment:
You can instantiate the List with the size you know you'll be using, to reserve enough space right away:
`new List<Embedding<float>>(data.Count);`

And if you just use a concrete List<...> type here (instead of IEnumerable)...